### ISSUE TITLE
Create Library Templates Error

### ISSUE THREAD

**Issue by ray1919** [miso-lims/miso-lims#1936]
I try to Create Library Templates in v0.2.176 & v0.2.177, both return error in browser inspect tool console, and the page does not response after click 'save'.

```
TypeError: Cannot read property 'replace' of undefined       header_script.js?version=0.2.175:20113
    at getDesign (header_script.js?version=0.2.175:8782)
    at Object.update (header_script.js?version=0.2.175:8880)
    at header_script.js?version=0.2.175:5841
    at Array.forEach (<anonymous>)
    at Object.makeTable (header_script.js?version=0.2.175:5837)
    at new?quantity=1&projectId=:198
    at Object.requestConfiguration (header_script.js?version=0.2.175:8798)
    at HTMLDocument.<anonymous> (new?quantity=1&projectId=:198)
    at l (jquery-1.8.3.min.js:2)
    at Object.fireWith [as resolveWith] (jquery-1.8.3.min.js:2)
 Uncaught TypeError: Cannot read property 'replace' of undefined   header_script.js?version=0.2.175:8782
    at getDesign (header_script.js?version=0.2.175:8782)
    at Object.update (header_script.js?version=0.2.175:8880)
    at header_script.js?version=0.2.175:5841
    at Array.forEach (<anonymous>)
    at Object.makeTable (header_script.js?version=0.2.175:5837)
    at new?quantity=1&projectId=:198
    at Object.requestConfiguration (header_script.js?version=0.2.175:8798)
    at HTMLDocument.<anonymous> (new?quantity=1&projectId=:198)
    at l (jquery-1.8.3.min.js:2)
    at Object.fireWith [as resolveWith] (jquery-1.8.3.min.js:2)
```

Then, I add a lib template record in database manually, and add this template to one project, project page echo the record of this library template. The sample propagate form could display template field, however drowdown list only shows `(None)`. Is this a bug?

**Comment by djcooke** [comment]
The first part looks like a bug. I haven't been able to reproduce the error though. Can you provide a few more details to help me come up with a fix?

1. Are you using plain or detailed sample mode?
2. Did you get to the Create Library Templates page from the Library Templates list page, or from a Project page?
3. Which fields were filled out in the table - not (Unspecified)?

For the second part, I suspect that you are using detailed sample mode and the sample you've propagated from is not compatible with the library design you chose in the template. If you edit the library template, it shows in the Design column which sample class is compatible.

Thanks

**Comment by ray1919** [comment]
1. plain mode
2. Both tested, same error
3. all fields (even I leave all fields blank & click save, no any msg return, page do not refresh)

I retest the page with clean env, both chrome and firefox receive same error in inspect console
```
wget https://github.com/miso-lims/miso-lims/archive/master.zip
unzip master.zip 'miso-lims-master/.docker/*'
mv miso-lims-master/.docker miso-lims-compose
rm -r master.zip miso-lims-master/
cd miso-lims-compose
export MISO_DB_USER=tgaclims && export MISO_DB=lims && export MISO_DB_PASSWORD_FILE=./.miso_db_password && MISO_TAG=latest
echo "changeme" > ./.miso_db_password
docker-compose -f demo.plain.yml up
```
Docker version 18.09.6, build 481bc77
Ubuntu 18.04

For the second part, I updated the Lib Template record as the same as what I manually filled out in the sample propagate page. And the dropdown list did not show the Lib Temp record.

**Comment by djcooke** [comment]
Thanks for the information. I think I have a fix and will be testing it tomorrow morning

**Comment by djcooke** [comment]
This turned out to be several bugs:

* Couldn't create library templates in plain sample mode
* Library template options failed to populate in plain sample mode
* Selection and strategy types were not set based on selected library template

All fixed in https://github.com/miso-lims/miso-lims/pull/1946 which will be included in release v0.2.179

**Comment by ray1919** [comment]
> This turned out to be several bugs:
> 
> * Couldn't create library templates in plain sample mode
> * Library template options failed to populate in plain sample mode
> * Selection and strategy types were not set based on selected library template
> 
> All fixed in #1946 which will be included in release v0.2.179

I tested the v0.2.179，these two bugs haven't been fixed?

save error as before

**Comment by djcooke** [comment]
I've retested v0.2.179 just now and the fixes seem to be working as intended. Are you sure you are running v0.2.179? It has not yet been officially released, and is due to be made official June 6 if no major bugs are found. This means that the `master` branch has not been updated yet, and the Docker images have not been pushed to Docker Hub. Note that official releases will have a change log posted on the releases page: https://github.com/miso-lims/miso-lims/releases

We recommend waiting until the release is official, but if you wish to use it before then, you can check out the tag and build the image from source as detailed in the Docker Compose guide under "Building and running from source": https://miso-lims.github.io/miso-lims/adm/compose-installation-guide

Closing for now, but please reopen the issue if you are sure you're running v0.2.179 and still experience these issues.
